File: submissions/cpp/55.cpp
class Solution {
public:
    bool canJump(vector<int> &nums) {
        int step = 0;
        for (int i = 0; i < nums.size() - 1; ++i) {
            int curr = i + nums[i];
            step = max(step, curr);
            if (step <= i)
                return false;
        }
        return true;
    }
};

File: LeetCode/python3/55.py
class Solution:
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        pos = 0
        for i in range(len(nums)-1):
            pos = max(pos, i + nums[i])
            if pos <= i:
                return False
        return True
